sherpa/optmethods/tests/test_optmethods.py

```diff
@@ -41,7 +41,7 @@ def tst_opt(opt, fct, npar, reltol=1.0e-3, abstol=1.0e-3):
     x0, xmin, xmax, fmin = init(fct.__name__, npar)
     status, x, fval, msg, xtra = opt(fct, x0, xmin, xmax)
     assert fmin == pytest.approx(fval, rel=reltol, abs=abstol)
-    if opt == lmdif:
+    if opt == lmdif and _ncpus > 1:
```

Comment:
@dtnguyen2 - so to check, the opt function will do something different if called with numcores>1 but only if the optimiser is lmdif?

Reply:
Yes.   The function ```test_opt``` in the file _sherpa/optmethods/tests/test_optmethods.py_ (probably should have been called ```test_lmdif_using_ncores```) was writen in PR #810 to test the parallelization of the Jacobian (derivatives with respect to the free parameters) in lmdif.   As you can see there is a check 
```
    if opt == lmdif and _ncpus > 1:
        status, x, fval, msg, xtra = opt(fct, x0, xmin, xmax, numcores=_ncpus)
        assert fmin  == pytest.approx(fval, rel=reltol, abs=abstol)
        assert xtra.get('num_parallel_map') != 0
```
that ```num_parallel_map``` is non-zero a variable which is only returned from lmdif, ie no other optimization method returns it.  The check was put in when @olaurino wanted to know how does one verify if the multicores were used or not.  The file _sherpa/optmethods/tests/test_ncoresopt.py_ contains the tests for the other  optimization methods (```DifEvo``` and ```NelderMead```)